src/screens/DiscoveryScreenV2/fragment/DomainFragment.js

```diff
@@ -82,43 +94,74 @@
     navigation.push('DomainScreen', navigationParam);
   };
 
-  const __handleFollow = async (from, willFollow, item, index) => {
+  const handleDomain = async (from, willFollow, item, index) => {
     if (from === FROM_FOLLOWED_DOMAIN_INITIAL) {
-      console.log('from 0');
       const newFollowedDomains = [...domains];
       newFollowedDomains[index].following = !!willFollow;
       newFollowedDomains[index].user_id_follower = myId;
 
-      console.log('newFollowedDomains', newFollowedDomains[index]);
-
-      FollowingAction.setFollowingDomain(newFollowedDomains, followingDispatch);
-      DiscoveryAction.setDiscoveryInitialDomains(newFollowedDomains, discoveryDispatch);
+      const newDomain = newFollowedDomains[index];
+      const newDomainList = discovery.initialDomains.map((domain) => {
+        if (domain.domain_page_id === newDomain.domain_page_id) {
+          return newDomain;
+        }
+        return domain;
+      });
+      FollowingAction.setFollowingDomain(newDomainList, followingDispatch);
+      DiscoveryAction.setDiscoveryInitialDomains(newDomainList, discoveryDispatch);
     }
     if (from === FROM_FOLLOWED_DOMAIN) {
-      console.log('from 1');
-      const newFollowedDomains = [...followedDomains];
+      const newFollowedDomains = [...newMapFollowedDomain];
       newFollowedDomains[index].following = !!willFollow;
       newFollowedDomains[index].user_id_follower = myId;
 
-      setFollowedDomains(newFollowedDomains);
+      const newDomain = newFollowedDomains[index];
+      setFollowedDomains(
+        followedDomains.map((domain) => {
+          if (domain.domain_name === newDomain.domain_name) {
+            return newDomain;
+          }
+          return domain;
+        })
+      );
     }
 
     if (from === FROM_UNFOLLOWED_DOMAIN) {
-      console.log('from 2');
-      const newUnfollowedDomains = [...unfollowedDomains];
+      const newUnfollowedDomains = [...newMapUnfollowedDomain];
       newUnfollowedDomains[index].following = !!willFollow;
       newUnfollowedDomains[index].user_id_follower = myId;
 
-      setUnfollowedDomains(newUnfollowedDomains);
+      const newDomain = newUnfollowedDomains[index];
+
+      setUnfollowedDomains(
+        unfollowedDomains.map((domain) => {
+          if (domain.domain_name === newDomain.domain_name) {
+            return newDomain;
+          }
+          return domain;
+        })
+      );
     }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `handleDomain` function has repetitive code blocks for each condition (`from === FROM_FOLLOWED_DOMAIN_INITIAL`, `from === FROM_FOLLOWED_DOMAIN`, `from === FROM_UNFOLLOWED_DOMAIN`). The repeated logic can be abstracted into a helper function to improve maintainability and readability.

```diff
+  const updateDomain = (domains, newDomains, setDomains) => {
+    const updatedDomains = domains.map((domain) => {
+      if (domain.domain_name === newDomains.domain_name) {
+        return newDomains;
+      }
+      return domain;
+    });
+    setDomains(updatedDomains);
+  };

-  const handleDomain = async (from, willFollow, item, index) => {
-    // ... existing code ...
-    if (from === FROM_FOLLOWED_DOMAIN_INITIAL) {
-      // ... existing code ...
-      FollowingAction.setFollowingDomain(newDomainList, followingDispatch);
-      DiscoveryAction.setDiscoveryInitialDomains(newDomainList, discoveryDispatch);
-    }
-    if (from === FROM_FOLLOWED_DOMAIN) {
-      // ... existing code ...
-      setFollowedDomains(
-        followedDomains.map((domain) => {
-          if (domain.domain_name === newDomain.domain_name) {
-            return newDomain;
-          }
-          return domain;
-        })
-      );
-    }
-    if (from === FROM_UNFOLLOWED_DOMAIN) {
-      // ... existing code ...
-      setUnfollowedDomains(
-        unfollowedDomains.map((domain) => {
-          if (domain.domain_name === newDomain.domain_name) {
-            return newDomain;
-          }
-          return domain;
-        })
-      );
-    }
-  };

+  const handleDomain = async (from, willFollow, item, index) => {
+    // ... existing code ...
+    if (from === FROM_FOLLOWED_DOMAIN_INITIAL) {
+      // ... existing code ...
+      updateDomain(discovery.initialDomains, newDomainList, DiscoveryAction.setDiscoveryInitialDomains);
+      FollowingAction.setFollowingDomain(newDomainList, followingDispatch);
+    }
+    if (from === FROM_FOLLOWED_DOMAIN) {
+      // ... existing code ...
+      updateDomain(followedDomains, newDomain, setFollowedDomains);
+    }
+    if (from === FROM_UNFOLLOWED_DOMAIN) {
+      // ... existing code ...
+      updateDomain(unfollowedDomains, newDomain, setUnfollowedDomains);
+    }
+  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->